torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -395,8 +395,8 @@ std::shared_ptr<c10::ivalue::Future> DistEngine::
       std::make_shared<c10::ivalue::Future>(c10::NoneType::get());
 
   futureGrads->addCallback(
-      [autogradContext, outputEdges, accumulateGradFuture, &futureGrads]() {
```

Comment:
This is possibly the worst offender: it is capturing a _reference_ to a _local variable_, which means this will pretty much always segfault unless when the callback is immediately called inline.